spec/integ/matrix-client-retrying.spec.js

```diff
@@ -52,15 +54,20 @@ describe("MatrixClient retrying", function() {
 
     });
 
-    it("should mark events as EventStatus.CANCELLED when cancelled", function(done) {
+    it("should mark events as EventStatus.CANCELLED when cancelled", function() {
         // send a couple of events; the second will be queued
-        client.sendMessage(roomId, "m1").then(function(ev) {
-            expect(ev).toEqual(ev1);
-        });
-        client.sendMessage(roomId, "m2").then(function(ev) {
-            expect(ev).toEqual(ev2);
+        const p1 = client.sendMessage(roomId, "m1").then(function(ev) {
+            // we expect the first message to fail
+            throw new Error('Message 1 unexpectedly sent successfully');
+        }, (e) => {
+            // this is expected
```

Comment:
It should do, because this message fails *before* it is cancelled. The promise is rejected when it fails.

OTOH that doesn't explain why the tests are failing. /me investigates